### ISSUE TITLE
Request: Use current location for weather forecast

### ISSUE THREAD

**Issue by bitplane** [MetaWatchOpenProjects/MWM-for-Android-Gen1#5]
I spend weekends in the North of England and week days in the South. Rather than me input my current location twice a week, it would be much nicer if it used the phone's current or last known location from Android's location API.

Also, rather than use Google's undocumented weather API (which only works with named cities), the app ought to use some other, more stable API (for example [Yahoo! Weather](http://developer.yahoo.com/weather/))

**Comment by laurielamberth** [comment]
I agree this needs to be added to the Metawatch/Metawatch Manager base code -- or created as an option to set local/fixed locations.  I, too, travel to a lot of different places during any given month...

**Comment by benjymous** [comment]
This is one of the enhancements I'm planning on adding - I just need to find a decent free/open geolocation web service that I can plug into a suitable weather api

geonames.org looks promising, but its free service is limited to 30k queries a day, so would eventually get swamped when too many people's watches start updating (1250 requests/hr) - I suppose I could make people sign up for their own free account, unless someone at metawatch wants to stump up for a commercial license?

I'll try and get something working this weekend, and will post back here

**Comment by bitplane** [comment]
Cool

There's no real need to resolve the location to a place name unless the GPS coordinates have changed by 30 miles or more, so I doubt we'd ever hit the 30k/day limit if the location name was cached.

**Comment by benjymous** [comment]
I'm chatting with one of the guys at http://www.wunderground.com - I've heard good things about them, and they have a nice all-inclusive API (so I wouldn't have to make multiple API calls across multiple sites to get all the data I need) they seem pretty positive about supporting OSS projects.

I've signed up for a free developer account, so will see how easily I can plug that in to the existing code.

**Comment by benjymous** [comment]
I can report I've got some prototype code working - It needs some good testing, and I need to work out the deal with the Weather Underground api key, but hopefully I should be able to release something soon.

**Comment by benjymous** [comment]
You can find my "RM2" build of my modified version of metawatch here:
http://www.metawatch.org/forums/thread/267/craig-s-metawatch-v1-1-4?page=3#dis-post-1606
(code in my fork here on github)
